### PR TITLE
Completed begin_intro use case

### PR DESCRIPTION
**Pull request by yunfeikevinwang**
Simple implementation of begin_intro use case. Add methods to game entity to get Villager/Werewolf Names.

### CODE REVIEW

**Review by daniyaalf — COMMENTED** [review]
I don't like having the Game object be in the Input Data / Output Data since the entity is getting too close to the interface adapters and could be modified by lower level modules. I think we should just write to a csv file (use a data access object) since the Game entity has only 5 attributes and they're very simple. It's really not that much work to recreate the Game entity even if it seems tedious at first. If we find a third option that's fine too.

File: src/use_case/begin_intro/BeginIntroOutputData.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package use_case.begin_intro;
+
+import entity.Game;
+import entity.PromptGenerator;
+
+public class BeginIntroOutputData {
+    private final String GPTMessage;
```
Comment (yhz3) [inline src/use_case/begin_intro/BeginIntroOutputData.java:7]: GPTMessage replace w introStory

File: src/use_case/begin_intro/BeginIntroOutputData.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    public BeginIntroOutputData(String GPTMessage) {
+        this.GPTMessage = GPTMessage;
```
Comment (yhz3) [inline src/use_case/begin_intro/BeginIntroOutputData.java:10]: here too

File: src/use_case/begin_intro/BeginIntroOutputData.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
+    public String getGPTMessage() {
+        return GPTMessage;
```
Comment (yhz3) [inline src/use_case/begin_intro/BeginIntroOutputData.java:14]: here too

File: src/use_case/begin_intro/BeginIntroOutputData.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import entity.PromptGenerator;
+
+public class BeginIntroOutputData {
+    private final String introStory;
+
+    public BeginIntroOutputData(String introStory) {
+        this.introStory = introStory;
+    }
+
+    public String getGPTMessage() {
```
Comment (yhz3) [inline src/use_case/begin_intro/BeginIntroOutputData.java:13]: getIntroStory

File: src/use_case/begin_intro/BeginIntroInteractor.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        String introStory = this.gptDataAccessObject.getResponse(prompt);
+
+        // Keep track of the ChatGPT response. User prompt is automatically stored when generating prompt.
+        this.promptGenerator.getConversationHistory().addGPTMessage(introStory);
+
+        // Save Game and PromptGenerator Data
+        this.gameDataAccessObject.save(this.game);
+        this.conversationDataAccessObject.save(this.promptGenerator);
+
+        BeginIntroOutputData response = new BeginIntroOutputData(introStory);
```
Comment (yhz3) [inline src/use_case/begin_intro/BeginIntroInteractor.java:47]: change response to beginIntroOutputData

**Review by yhz3 — APPROVED** [review]

**Pull request merged**